app/editor/src/features/admin/data-locations/DataLocationForm.tsx

```diff
@@ -45,7 +45,7 @@ const DataLocationForm: React.FC = () => {
   );
 
   const dataLocationId = Number(id);
-  const connectionOptions = getSortableOptions(connections, [
+  const connectionOptions = getSortableOptions(connections, dataLocation.connectionId, [
```

Comment:
Had to do this to fix the bug where the currently selected value has been disabled and it wasn't been included in the list.